### PR TITLE
iox-#542 use the signal handler everywhere

### PR DESCRIPTION
**Pull request by elfenpiff**
## Pre-Review Checklist for the PR Author

1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## Post-review Checklist for the Eclipse Committer

1. [x] All checkboxes in the PR checklist are checked or crossed out
1. [x] Merge

## References

- #542

### CODE REVIEW

**Review by dkroenke — APPROVED** [review]

File: iceoryx_examples/icedelivery_in_c/ice_c_publisher.c [hunk 1]
```diff
@@ -77,6 +77,7 @@ void sending()
 int main()
 {
     signal(SIGINT, sigHandler);
+    signal(SIGTERM, sigHandler);
```
Comment (elBoberido) [inline iceoryx_examples/icedelivery_in_c/ice_c_publisher.c:80]: okay, this is C code. Maybe we need a issue or extend #542 to replace these with `sigaction`. Could also be just a free function like `registerSignalHandler` but without the restoring of the previous signal handler

File: iceoryx_examples/waitset/ice_waitset_publisher.cpp [hunk 2]
```diff
@@ -49,7 +51,7 @@ void sending()
 
 int main()
 {
-    signal(SIGINT, sigHandler);
+    auto signalGuard = iox::posix::registerSignalHandler(iox::posix::Signal::INT, sigHandler);
```
Comment (elBoberido) [inline iceoryx_examples/waitset/ice_waitset_publisher.cpp:54]: please register the signal handler either in `sending` or here. I suggest to do it in `main`

**Review by elBoberido — COMMENTED** [review]
nice reduction of code at some places which results in better readability

File: iceoryx_examples/icedelivery_in_c/ice_c_publisher.c (hunk 1 above)
Comment (elfenpiff) [reply iceoryx_examples/icedelivery_in_c/ice_c_publisher.c:80]: I added this as a todo

**Review by dkroenke — DISMISSED** [review]

File: iceoryx_posh/source/roudi/memory/posix_shm_memory_provider.cpp [hunk 3]
```diff
@@ -1,4 +1,5 @@
-// Copyright (c) 2020 by Robert Bosch GmbH, Apex.AI Inc. All rights reserved.
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2020 2021 by Apex.AI Inc. All rights reserved.
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_posh/source/roudi/memory/posix_shm_memory_provider.cpp:2]: There is a dash missing.

File: iceoryx_examples/waitset/ice_waitset_publisher.cpp [hunk 4]
```diff
@@ -31,8 +31,6 @@ static void sigHandler(int f_sig [[gnu::unused]])
 
 void sending()
 {
-    iox::runtime::PoshRuntime::initRuntime("iox-ex-publisher-waitset");
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_examples/waitset/ice_waitset_publisher.cpp:34]: Why is this removed? Without ``initRuntime`` the example should't work...

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

**Review by dkroenke — APPROVED** [review]

**Review by elBoberido — APPROVED** [review]
